http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java

```diff
@@ -230,6 +292,7 @@ public DefaultHttpClient(@Parameter LoadBalancer loadBalancer,
     public DefaultHttpClient(@Parameter LoadBalancer loadBalancer,
                              @Parameter HttpClientConfiguration configuration,
                              @Parameter @Nullable String contextPath,
+                             @Parameter @Nullable AnnotationMetadata annotationMetadata,
```

Comment:
Whilst I see what you are trying to do here, add `AnnotationMetadata` do the client is not the right approach since this introduces coupling between the introduction advice and the low-level client which are really separate things.

If what we want to allow is customization of how filters are resolving I believe the correct thing would be to introduction a constructor argument that takes the `FilterResolver` and pass that in from the introduction advice. The default `FilterResolver` would in that case just return the list of filters without applying any other filters.

Reply:
That makes sense to me. I will change